plane/src/drone/docker/commands.rs

```diff
@@ -113,13 +119,52 @@ pub async fn get_port(docker: &Docker, container_id: &ContainerId) -> Result<u16
     Err(anyhow::anyhow!("Failed to get port after retries."))
 }
 
+// Canonicalize a path without talking to the filesystem.
+//
+// Copy-pasted this function from Cargo.
+// https://github.com/rust-lang/cargo/blob/fede83ccf973457de319ba6fa0e36ead454d2e20/src/cargo/util/paths.rs#L61
+//
+// The issue is Path's canonicalize function contacts the filesystem and checks
+// if the path exists. Instead, we effectively want a string-operation.
+//
+// We use this function before passing a directory to Docker's "Binds" field,
+// which works similarly to "Mounts", but Docker also creates the directory if
+// it does not exist.
+pub fn normalize_path(path: &Path) -> PathBuf {
```

Comment:
Instead of normalizing the path, thoughts on just validating that the portion included in the spawn request is valid? You could take the same approach of iterating over `components()`, and just fail a URL if it includes anything other than `Normal` and `Prefix` parts.

I'm trying to think of an example where someone would want to use `../` for their own mount, and it's hard to think of one.

Reply:
Sounds good, makes sense! 

> I'm trying to think of an example where someone would want to use ../ for their own mount, and it's hard to think of one.

Just to confirm, you're saying here that a spawn with a `mount` like `"custom-mount/../different-folder/././subfolder/../../actual-folder"` (which technically would be a subfolder of `mount_base`) is ok to not allow? I think that's reasonable -- I also can't think of a scenario where someone would do that when they could do a much simpler direct path (`"actual-folder"`). 

Calling this out because I'm removing this test case.